### PR TITLE
A simple service to check whether disk full in /var/lib/glusterd

### PR DESCRIPTION
**Pull request by SaravanaStorageNetwork**
stop glusterd service, if so.

Signed-off-by: Saravanakumar Arumugam <sarumuga@redhat.com>

### CODE REVIEW

**Comment by SaravanaStorageNetwork** [comment]
@humblec @jarrpa @raghavendra-talur @obnoxxx PTAL   - This PR does not need changes in templates - it runs inside container and stop glusterd service if  space in /var/lib/glusterd < 99%   - this will in turn help existing liveness probe to fail.

File: CentOS/gluster-check-size.service [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Check glusterd config directory full
+After=glusterd.service
+
+[Service]
+Type=simple
+ExecStart=/usr/local/bin/readiness_probe.sh
```
Comment (phlogistonjohn) [inline CentOS/gluster-check-size.service]: check_size.sh or readiness_probe.sh -- this PR creates the former but this line refers to the latter.

File: CentOS/check_size.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# Disk full
+while true
+do
+        # sleep early to get glusterd settled
+        sleep 120
+        current_usage=$( df --output=pcent  '/var/lib/glusterd' | tail -n1  | awk {'print $1'} )
+        max_usage=99%
+        if [ "${current_usage%?}" -ge "${max_usage%?}" ]; then
+                echo "running out of space in /var/lib/glusterd - stop glusterd"
```
Comment (raghavendra-talur) [inline CentOS/check_size.sh]: I would just like to see this message changed to include:
1. the owner of log, that is the script name
2. s/stop/stopping
3. add another log after stopping to say it is stopped
4. add time to all the logs.

File: CentOS/check_size.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Disk full
+while true
+do
+        # sleep early to get glusterd settled
+        sleep 120
```
Comment (phlogistonjohn) [inline CentOS/check_size.sh]: can we set this up via an environment variable (with a default value)?

File: CentOS/gluster-check-size.service (hunk 1 above)
Comment (SaravanaStorageNetwork) [reply CentOS/gluster-check-size.service]: oops - fixing it

File: CentOS/check_size.sh (hunk 3 above)
Comment (SaravanaStorageNetwork) [reply CentOS/check_size.sh]: Using environment variable in systemd inside the script seems little tricky - will update this later

**Comment by SaravanaStorageNetwork** [comment]
@humblec @jarrpa @raghavendra-talur @obnoxxx PTAL  - addressed review comments

File: CentOS/Dockerfile [hunk 4]
```diff
@@ -59,18 +59,23 @@ ADD gluster-block-setup.service /etc/systemd/system/gluster-block-setup.service
 ADD gluster-block-setup.sh /usr/sbin/gluster-block-setup.sh
 ADD update-params.sh /usr/local/bin/update-params.sh
 ADD tcmu-runner-params /etc/sysconfig/tcmu-runner-params
+ADD gluster-check-size.service  /etc/systemd/system/gluster-check-size.service
```
Comment (humblec) [inline CentOS/Dockerfile]: may be check_diskspace looks better to me .

Comment (SaravanaStorageNetwork) [reply CentOS/Dockerfile]: Sure - updating

File: CentOS/gluster-check-diskspace.service [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Check glusterd config directory full
+After=glusterd.service
+
+[Service]
+Type=simple
+ExecStart=/usr/local/bin/check_size.sh
```
Comment (Madhu-1) [inline CentOS/gluster-check-diskspace.service]: is this file present `/usr/local/bin/check_size.sh`

Comment (SaravanaStorageNetwork) [reply CentOS/gluster-check-diskspace.service]: good catch - updating :)

File: CentOS/check_size.sh (hunk 3 above)
Comment (phlogistonjohn) [reply CentOS/check_size.sh]: Is the parameter https://www.freedesktop.org/software/systemd/man/systemd.exec.html#PassEnvironment=  not available in the version of systemd used?

**Comment by humblec** [comment]
LGTM. Merging !

**Comment by phlogistonjohn** [comment]
@humblec, I am not sure if you were aware but I still had a nack on this. I was hoping to finish the discussion about the environment variables before I changed my review status. The change looks fine outside of my concerns about the hard coded timing, but I would have liked that conversation to have come to a conclusion first.

**Comment by jarrpa** [comment]
I agree, the hard-coded 120 is arbitrary and may artificially prolong the time it takes for the check to kill the pod. Also, just a set sleep timer seems wrong in general. It would be better to loop over a check of glusterd's actual availability and then check disk space. Or hell, do you even need to wait for glusterd? Couldn't you just check if /var/lib/glusterd exists (which it should)?

**Comment by humblec** [comment]
@phlogistonjohn while I had acks from @Madhu-1 @raghavendra-talur and based on the positive result of testing which @SaravanaStorageNetwork performed at his end, I went ahead, but yes I could have waited for your ack as well. The only comment which was not addressed here in this patch was `sleep` to `env` variable conversion, which @SaravanaStorageNetwork agreed to try out and send in another PR. In short, I was aware that this conversion didnt happen in this PR.

**Comment by humblec** [comment]
```It would be better to loop over a check of glusterd's actual availability and then check disk space```. Its not required as we already have a probe on glusterd status ie if its down OCP will respin container any way.

**Comment by jarrpa** [comment]
Then why are we even waiting for 120 seconds? If the disk is full why wouldn't we check that right away?

**Comment by SaravanaStorageNetwork** [comment]
> Then why are we even waiting for 120 seconds? If the disk is full why wouldn't we check that right away?

The disk may get full due the reason that volumes keep getting created and this specific partition(all gluster configuration) getting filled due to this.

**Comment by jarrpa** [comment]
Maybe put the sleep at the end, then, so that at least you check the disk right at startup and can fail early?

**Pull request merged**